contracts/pair_concentrated/src/queries.rs

```diff
@@ -192,10 +192,24 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
             };
 
             // Calculate average from buy and sell prices
-            let spot_price = (get_spot_price(0)? + get_spot_price(1)?) / TWO;
+            let mut spot_price = (get_spot_price(0)? + get_spot_price(1)?) / TWO;
+
+            // Denormalize the price
+            let quote_precision = precisions
+                .get_precision(&AssetInfo::native(&quote_asset_denom))
+                .map_err(|err| StdError::generic_err(err.to_string()))?;
+            let base_precision = precisions
+                .get_precision(&AssetInfo::native(&base_asset_denom))
+                .map_err(|err| StdError::generic_err(err.to_string()))?;
+            spot_price *= Decimal256::from_ratio(
+                10u128.pow(quote_precision.into()),
+                10u128.pow(base_precision.into()),
+            );
```

Comment:
Please consider some error handling if the scaling factor application truncates the result to zero.

E.g. the original spot price is not zero and the final one is zero => throw an error

Reply:
Thanks! Added error handling and one more test for this case